libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -212,11 +221,43 @@ bool IRGeneratorForStatements::visit(TupleExpression const& _tuple)
 		solUnimplementedAssert(false, "");
 	else
 	{
-		solUnimplementedAssert(!_tuple.annotation().lValueRequested, "");
-		solUnimplementedAssert(_tuple.components().size() == 1, "");
-		solAssert(_tuple.components().front(), "");
-		_tuple.components().front()->accept(*this);
-		defineExpression(_tuple) << m_context.variable(*_tuple.components().front()) << "\n";
+		bool lValueRequested = _tuple.annotation().lValueRequested;
+		if (lValueRequested)
+			solAssert(!m_currentLValue, "");
+		if (_tuple.components().size() == 1)
```

Comment:
Can these two cases be a little combined?
Why does the first case not modify `m_currentLValue`?

Reply:
A one-element tuple expression doesn't have tuple type, does it? And therefore the variables can't have ``_component_1`` suffices and the lvalue has to be inherited from the contained element (that's why it's not modified, but it is just verified, that the component expression has set it)... and given all that I don't think the cases can be reasonably combined...

Reply:
I should add explicit tests, though - doing that now.